### PR TITLE
Don't look up services in all namespaces.

### LINKED ISSUE #2495: Skaffold dev fails when user cannot list deployments at cluster scope
The deployment status check from Skaffold tries to list all deployments at cluster scope, when running Skaffold without the `--namespace` flag. For example:

```
cd examples/nodejs
kubectl auth can-i list deploy --all-namespaces  # -> no
skaffold dev
```

Fails with 
```
FATA[0001] exiting dev mode because first deploy failed: could not fetch deployments: could not fetch deployments: deployments.apps is forbidden: User "default-user" cannot list resource "deployments" in API group "apps" at the cluster scope
```
On the other hand:
```
kubectl auth can-i list deploy -n default  # -> yes
skaffold dev -n default
```
works as expected.

---

The reason for this is that `pkg/skaffold/deploy/status_check.go:80` does 
```go
client.AppsV1().Deployments(ns).List(metav1.ListOptions{
		LabelSelector: l.K8sManagedByLabelKeyValueString(),
	})
```
with `ns==""` (due to not specifying a namespace), and that means _all namespaces_.

### PR DESCRIPTION
**Pull request by tejal29**
fixes #2495, 

In #2640, i made changes to collect all namespaces for deployed resources.
In PR, we go through all namespaces in the context and then look for services in each namespace.

### CODE REVIEW

**Comment by tejal29** [comment]
hmm, for user defined port forwarding, why do we need to fetch it? We should be able to get the specific resource right?

**Comment by priyawadhwa** [comment]
ah that's true, nevermind!

File: pkg/skaffold/kubernetes/portforward/resource_forwarder_test.go [hunk 1]
```diff
@@ -252,3 +259,101 @@ func TestUserDefinedResources(t *testing.T) {
...
+						Namespace: "test",
+						Labels: map[string]string{
+							deploy.RunIDLabel: "9876-6789",
+						},
+					},
+					Spec: v1.ServiceSpec{Ports: []v1.ServicePort{{Port: 8080}}},
+				},
+			},
+		}, {
+			description: "services present but does not expose any port",
```
Comment (priyawadhwa) [inline pkg/skaffold/kubernetes/portforward/resource_forwarder_test.go]: is this possible for a service?

Comment (tejal29) [reply pkg/skaffold/kubernetes/portforward/resource_forwarder_test.go]: you wont normally do it. But yes, you can not have any ports defined in the Spec and the service is created.

```
$ kubectl apply -f microservices/leeroy-app/kubernetes/
service/leeroy-app created
deployment.apps/leeroy-app created

$ git diff microservices/leeroy-app/kubernetes/
diff --git a/examples/microservices/leeroy-app/kubernetes/deployment.yaml b/examples/microservices/leeroy-app/kubernetes/deployment.yaml
index 56ef7152..46feab6d 100644
--- a/examples/microservices/leeroy-app/kubernetes/deployment.yaml
+++ b/examples/microservices/leeroy-app/kubernetes/deployment.yaml
@@ -6,9 +6,6 @@ metadata:
     app: leeroy-app
 spec:
   clusterIP: None
-  ports:
-    - port: 50051
-      name: leeroy-app
   selector:
     app: leeroy-app
 ---
tejaldesai@@examples (move_resolve_namespace)$ 
```

**Pull request merged**